impeller/renderer/backend/gles/buffer_bindings_gles.cc

```diff
@@ -320,65 +396,44 @@ bool BufferBindingsGLES::BindUniformBuffer(const ProcTableGLES& gl,
-                          element_count,  // count
-                          buffer_data     // data
-            );
-            continue;
-          case sizeof(Vector2):
-            gl.Uniform2fv(location,       // location
-                          element_count,  // count
-                          buffer_data     // data
-            );
-            continue;
-          case sizeof(Scalar):
-            gl.Uniform1fv(location,       // location
-                          element_count,  // count
-                          buffer_data     // data
-            );
-            continue;
-        }
-        VALIDATION_LOG << "Size " << member.size
-                       << " could not be mapped ShaderType::kFloat for key: "
-                       << member.name;
-      case ShaderType::kBoolean:
-      case ShaderType::kSignedByte:
-      case ShaderType::kUnsignedByte:
-      case ShaderType::kSignedShort:
-      case ShaderType::kUnsignedShort:
-      case ShaderType::kSignedInt:
-      case ShaderType::kUnsignedInt:
-      case ShaderType::kSignedInt64:
-      case ShaderType::kUnsignedInt64:
-      case ShaderType::kAtomicCounter:
-      case ShaderType::kUnknown:
-      case ShaderType::kVoid:
-      case ShaderType::kHalfFloat:
-      case ShaderType::kDouble:
-      case ShaderType::kStruct:
-      case ShaderType::kImage:
-      case ShaderType::kSampledImage:
-      case ShaderType::kSampler:
-        VALIDATION_LOG << "Could not bind uniform buffer data for key: "
-                       << member.name << " : " << static_cast<int>(member.type);
-        return false;
+    if (member.type != ShaderType::kFloat) {
+      VALIDATION_LOG << "Could not bind uniform buffer data for key: "
+                     << member.name << " : " << static_cast<int>(member.type);
+      return false;
+    }
+
+    switch (member.size) {
+      case sizeof(Matrix):
+        gl.UniformMatrix4fv(location,       // location
+                            element_count,  // count
+                            GL_FALSE,       // normalize
+                            buffer_data     // data
+        );
+        continue;
+      case sizeof(Vector4):
+        gl.Uniform4fv(location,       // location
+                      element_count,  // count
+                      buffer_data     // data
+        );
+        continue;
+      case sizeof(Vector3):
+        gl.Uniform3fv(location,       // location
+                      element_count,  // count
+                      buffer_data     // data
+        );
+        continue;
+      case sizeof(Vector2):
+        gl.Uniform2fv(location,       // location
+                      element_count,  // count
+                      buffer_data     // data
+        );
+        continue;
+      case sizeof(Scalar):
+        gl.Uniform1fv(location,       // location
+                      element_count,  // count
+                      buffer_data     // data
+        );
+        continue;
```

Comment:
Perhaps add the `default:` case that logs the member type so we don't silently ignore stuff? But yeah, at this point, probably all types have shaken out.

Reply:
Done